### PR TITLE
chore: Updating dependabot to support Javascript, Typescript and docker

### PR DESCRIPTION
**Pull request by rajpalc7**

### CODE REVIEW

**Comment by rajpalc7** [comment]
Hi @WadeBarnes , if this looks good , I can squash and merge.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -4,9 +4,81 @@
...
+  - package-ecosystem: "npm"
+    directory: "/cypress"
+    schedule:
+      interval: "weekly"
+      day: "monday"
+      time: "04:00"
+      timezone: "Canada/Pacific"
+    ignore:
+      - dependency-name: "*"
+        update-types: ["version-update:semver-major"]
```
Comment (WadeBarnes) [inline .github/dependabot.yml:42]: I don't see any dependency manifests in the  `/cypress` folder.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -4,9 +4,81 @@
...
+  - package-ecosystem: "npm"
+    directory: "/server/src"
+    schedule:
+      interval: "weekly"
+      day: "monday"
+      time: "04:00"
+      timezone: "Canada/Pacific"
+    ignore:
+      - dependency-name: "*"
+        update-types: ["version-update:semver-major"]
```
Comment (WadeBarnes) [inline .github/dependabot.yml:66]: I don't see any dependency manifests in the `/server/src` folder.

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
In addition to the items mentioned below, there is also a dockerfile in the root `/` directory.

**Comment by rajpalc7** [comment]
Thanks @WadeBarnes I have made changes as per your suggestion.

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
@rajpalc7, You were a little to aggressive with deleting the `npm` update sections in your last commit right before you rebased the commits.  You deleted one that was needed.

**Comment by rajpalc7** [comment]
Thanks @WadeBarnes , I have added the deleted one now.

**Review by WadeBarnes — APPROVED** [review]
Looks good to me.  I'll let @wadeking98 do a final review before merging.

File: .github/dependabot.yml [hunk 3]
```diff
@@ -1,12 +1,77 @@
...
+      day: "monday"
+      time: "04:00"
+      timezone: "Canada/Pacific"
+    ignore:
+      - dependency-name: "*"
+        update-types: ["version-update:semver-major", "version-update:semver-patch"]
+
+  # Maintain dependencies for TypeScript and JavaScript
+  - package-ecosystem: "npm"
+    directory: "/client"
```
Comment (cvarjao) [inline .github/dependabot.yml:30]: `/client` is a submodule, and its lock file is managed top level. Please remove `/client` entry.

File: .github/dependabot.yml [hunk 4]
```diff
@@ -1,12 +1,77 @@
...
+      day: "monday"
+      time: "04:00"
+      timezone: "Canada/Pacific"
+    ignore:
+      - dependency-name: "*"
+        update-types: ["version-update:semver-major", "version-update:semver-patch"]
+
+  # Maintain dependencies for TypeScript and JavaScript
+  - package-ecosystem: "npm"
+    directory: "/server"
```
Comment (cvarjao) [inline .github/dependabot.yml:42]: `/server` is a submodule, and its lock file is managed top level. Please remove `/server` entry.

**Review by cvarjao — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 4 above)
Comment (rajpalc7) [reply .github/dependabot.yml]: Changes have been made

File: .github/dependabot.yml (hunk 3 above)
Comment (rajpalc7) [reply .github/dependabot.yml]: Changes have been made

**Review by cvarjao — APPROVED** [review]

**Pull request merged**